jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -548,59 +550,80 @@ protected void sendError(int code, String reason)
      * bypassing the {@link ErrorHandler} mechanisms and the response OutputStream.</p>
      *
-     * @param failure the Throwable that caused the problem
+     * @param th the Throwable that caused the problem
      */
-    protected void handleException(Throwable failure)
+    protected void handleException(Throwable th)
     {
-        // Unwrap wrapping Jetty exceptions.
-        if (failure instanceof RuntimeIOException)
-            failure = failure.getCause();
+        // Unwrap wrapping Jetty and Servlet exceptions.
+        Throwable cause = unwrap(th);
 
-        if (failure instanceof QuietException || !getServer().isRunning())
+        if (cause instanceof QuietException || !getServer().isRunning())
         {
             if (LOG.isDebugEnabled())
-                LOG.debug(_request.getRequestURI(), failure);
+                LOG.debug(_request.getRequestURI(), th);
         }
-        else if (failure instanceof BadMessageException | failure instanceof IOException | failure instanceof TimeoutException)
+        else if (cause instanceof BadMessageException | cause instanceof IOException | cause instanceof TimeoutException)
         {
             // No stack trace unless there is debug turned on
-            LOG.warn("{} {}",_request.getRequestURI(), failure.toString()); 
+            LOG.warn("{} {}",_request.getRequestURI(), cause.toString()); 
             if (LOG.isDebugEnabled())
-                LOG.debug(_request.getRequestURI(), failure);
+                LOG.debug(_request.getRequestURI(), th);
         }
         else
         {
-            LOG.warn(_request.getRequestURI(), failure);
+            LOG.warn(_request.getRequestURI(), th);
         }
 
         try
         {
-            _state.onError(failure);
+            _state.onError(th);
         }
         catch (Throwable e)
         {
-            if (e != failure)
-                failure.addSuppressed(e);
-            LOG.warn("ERROR dispatch failed", failure);
+            if (e != th)
+                th.addSuppressed(e);
+            LOG.warn("ERROR dispatch failed", th);
             // Try to send a minimal response.
-            minimalErrorResponse(failure);
+            minimalErrorResponse(cause);
         }
     }
 
-    private void minimalErrorResponse(Throwable failure)
+
+    protected Throwable unwrap(Throwable th)
+    {
+        // Unwrap to root cause:
+        // RuntimeIOExceptions are unwrapped as they are holders of IOExceptions that may be verbosity controlled
+        // ServletExceptions are unwrapped as are just general holders of exceptions
+        // UnavailableException are not unwrapped as they are specialized ServletExceptions that carry extra information.
+        while (th.getCause()!=null && (th instanceof RuntimeIOException || ((th instanceof ServletException) && !(th instanceof UnavailableException))))
+            th = th.getCause();
+        return th;
+    }
+    
+    private void minimalErrorResponse(Throwable th)
```

Comment:
This was `Throwable failure`, but you changed the parameter name and essentially rewrote the entire class.
Other change in this PR have you changing `Throwable x` to `Throwable failure`.
Seems inconsistent.